src/common/lib/client/realtimechannel.ts

```diff
@@ -189,7 +189,14 @@ class RealtimeChannel extends Channel {
       return false;
     }
     if (options?.params) {
-      if (!this.params || !Utils.shallowEquals(this.params, options.params)) {
+      // Don't check against the `agent` param - it isn't returned in the ATTACHED message
```

Comment:
I'm not sure I understand what's going on here, could you explain please? The relevant spec point is [RTS3c1](https://sdk.ably.com/builds/ably/specification/main/features/#RTS3c1), which doesn't mention any special case for the `agent` param. And I don't think I understand why it's relevant that `agent` "it isn't returned in the ATTACHED message".

Reply:
When you call `Channels.get` to get an already-attached channel, the client will compare the requested params with the params it received in the `ATTACHED` message to determine whether it would need to re-attach. Since realtime currently doesn't send the `agent` param in the `ATTACHED` message, the client will always believe that it needs to re-attach if the `agent` param is passed in to `Channels.get` but in reality the client will never actually need to re-attach because of the `agent` param, so this change just makes the client ignore this param from the check.